lib/inspec/plugin/v2/loader.rb

```diff
@@ -158,7 +158,7 @@ def list_managed_gems
     end
 
     def self.find_gem_directory(gem_name, version = nil)
-      matching_gem_versions = list_managed_gems.filter { |g| g.name == gem_name }.sort(&:version)
+      matching_gem_versions = list_managed_gems.filter { |g| g.name == gem_name }.sort_by(&:version)
```

Comment:
this is the ideal logic am looking for activation as well.. is it possible to bring similar logic as well while deleting different version specs?

Reply:
I think having this version check makes sense if it is absolutely necessary to delete a specific version. In this scenario the only need is to remove old cached activated resource pack gems.

Reply:
we do need this to activate a particular version too